frontend/main.go

```diff
@@ -145,16 +146,24 @@ func main() {
 		log.Fatalf("Error creating Kubernetes client: %s", err)
 	}
 
-	// Start server
-	r, err := startHTTPServer(&flags)
-	if err != nil {
-		log.Fatalf("Error starting server: %s", err)
-	}
-
-	log.Println("Starting Odigos UI...")
-	log.Printf("Odigos UI is available at: http://%s:%d", flags.Address, flags.Port)
-	err = r.Run(fmt.Sprintf("%s:%d", flags.Address, flags.Port))
-	if err != nil {
-		log.Fatalf("Error starting server: %s", err)
+	// Attempt to start the server on port 3000 and increment the port number if it's already in use
+	for {
```

Comment:
If the err we encounter is not related to the port being bound already, then we might run this loop many times before something will break us out.

Perhaps we can limit the number of tested ports to something like 5 or 10? this way, if there is an issue - we do not iterate dozens of ports.

Reply:
limited the number of testing ports to 10